rir/src/compiler/pir/instruction.cpp

```diff
@@ -218,6 +218,42 @@ void Instruction::printGraph(std::ostream& out, bool tty) const {
     printEnv(out, tty);
 }
 
+bool Instruction::willDefinitelyNotOverflow() const {
+    if (!Add::Cast(this) && !Sub::Cast(this)) {
+        return false;
+    }
+    UsesTree uses(bb()->owner);
```

Comment:
please no expensive analysis in instructions! if you want to run an analysis then put it into opt/types.